nvaccess/nvda
# Unset systray icon and menu after destruction

### Link to issue number:

None

### Summary of the issue:

NVDA crashes when exiting with the speech viewer open.

With the following relevant stacktrace:

```python
Python stack for thread 14220 (MainThread):
  File "C:\work\repo\nvda\4\\source\nvda.pyw", line 266, in <module>
    core.main()
  File "C:\work\repo\nvda\4\source\core.py", line 662, in main
    app.MainLoop()
  File "C:\work\repo\nvda\4\.venv\lib\site-packages\wx\core.py", line 2237, in MainLoop
    rv = wx.PyApp.MainLoop(self)
  File "C:\work\repo\nvda\4\source\gui\__init__.py", line 696, in onOk
    core.triggerNVDAExit()
  File "C:\work\repo\nvda\4\source\core.py", line 235, in triggerNVDAExit
    preNVDAExit.notify()
  File "C:\work\repo\nvda\4\source\extensionPoints\__init__.py", line 47, in notify
    callWithSupportedKwargs(handler, **kwargs)
  File "C:\work\repo\nvda\4\source\extensionPoints\util.py", line 170, in callWithSupportedKwargs
    return func(*boundArguments.args, **boundArguments.kwargs)
  File "C:\work\repo\nvda\4\source\core.py", line 285, in _closeAllWindows
    wx.Yield()  # creates a temporary event loop and uses it instead to process pending messages
  File "C:\work\repo\nvda\4\source\speechViewer.py", line 111, in onDestroy
    self.onDestroyCallBack()
  File "C:\work\repo\nvda\4\source\speechViewer.py", line 177, in _cleanup
    _setActive(False)
  File "C:\work\repo\nvda\4\source\speechViewer.py", line 147, in _setActive
    gui.mainFrame.onSpeechViewerEnabled(isNowActive)
  File "C:\work\repo\nvda\4\source\gui\__init__.py", line 270, in onSpeechViewerEnabled
  ```

### Description of how this pull request fixes the issue:

Sets the value of sysTrayIcon and menu to `None` after processing their destruction earlier. Now the checks of `if sysTrayIcon` when closing windows are accurate, and the objects are destroyed fully before closing windows.

### Testing strategy:

Manual testing

- Open speech viewer (NVDA + n, t, s)
- Quit NVDA (NVDA + q, enter)
- check if NVDA crashes

General testing of exiting NVDA.

Existing NVDA system test smoke tests for exiting NVDA.

### Known issues with pull request:

None

### Change log entries:

None. regression

### Code Review Checklist:
